docs/development/setup.md

```diff
@@ -0,0 +1,160 @@
+
+4. Once the shoot is successfully reconciled by your local `gardenlet`, get its kubeconfig by executing:
+
+    ```shell script
+    kubectl get secret <prefix>-gcp-vmx.kubeconfig -n garden-dev -o yaml | yq r - 'data.kubeconfig' | base64 -d > dev/kubeconfig-gcp-vmx.yaml
+    ```
+   
+5. Install kubevirt and CDI in this cluster by executing the [install-kubevirt.sh](scripts/install-kubevirt.sh) script:
+
+    ```shell script
+    export KUBECONFIG=dev/kubeconfig-gcp-vmx.yaml
+    docs/development/scripts/install-kubevirt.sh
+    ```
+
+## Testing the Gardener Reconciliation Flow
+
+To test the Gardener reconciliation flow with the Kubevirt provider extensions, create the Kubevirt shoot cluster in your local `dev` project, by following these steps:
+
+1. Create the Kubevirt cloud profile, for example [cloudprofile-kubevirt.yaml](manifests/cloudprofile-kubevirt.yaml).
+
+    ```shell script
+    kubectl apply -f dev/cloudprofile-kubevirt.yaml
+    ```
+
+2. Create the shoot secret and secret binding. You should create a secret containing the kubeconfig for your provider cluster, and a corresponding secret binding:
+
+    ```shell script
+    kubectl create secret generic kubevirt-credentials -n garden-dev --from-file=kubeconfig=dev/kubeconfig-gcp-vmx.yaml
+    kubectl apply -f dev/secretbinding-kubevirt-credentials.yaml
+    ```
+
+3. Create the Kubevirt shoot itself. See [shoot-kubevirt.yaml](manifests/shoot-kubevirt.yaml) as an example. Note that the nodes CIDR for this shoot must be the same range as the pods CIDR of your provider cluster.
+
+    ```shell script
+    kubectl apply -f dev/shoot-kubevirt.yaml
+    ```
+   
+4. During the shoot reconciliation by your local `gardenlet`, you may want to:
+
+    * Monitor the `gardenlet` logs in your local console where `gardenlet` is running.
+    * Connect to the seed to monitor the shoot namespace `shoot--dev--kubevirt` and the logs of the Kubevirt provider extension in the `extension-provider-kubevirt-*` namespace.
+    * Connect to the provider cluster to monitor the `default` namespace where VMs and VMIs are being created.
+
+5. Once the shoot has been successfully reconciled, get its kubeconfig by executing:
+
+    ```shell script
+    kubectl get secret kubevirt.kubeconfig -n garden-dev -o yaml | yq r - 'data.kubeconfig' | base64 -d > dev/kubeconfig-kubevirt.yaml
+    ```
+   
+    At this point, you may want to connect to the Kubevirt shoot and check if it's usable.
+    
+## Testing the Gardener Deletion Flow
+
+To test the Gardener deletion flow with the Kubevirt provider extensions, delete the Kubevirt shoot cluster in your local `dev` project, by following these steps:
+
+1. Delete the Kubevirt shoot itself using the [delete](https://github.com/gardener/gardener/blob/master/hack/usage/delete) script.
+
+    ```shell script
+    ../gardener/hack/usage/delete shoot kubevirt garden-dev
+    ```
+   
```

Comment:
To avoid troubles with the path to the local copy of gardener/gardener, you can just use the commands
```bash
kubectl annotate shoot kubevirt -n garden-dev confirmation.gardener.cloud/deletion=1
kubectl delete shoot kubevirt -n garden-dev
```

Reply:
Done